src/main/java/pl/touk/sputnik/review/Review.java

```diff
@@ -36,11 +37,11 @@
     private List<String> messages = new ArrayList<>();
     private Map<String, Short> scores = new HashMap<>();
 
-    private ReviewFormatter formatter;
+    private final ReviewFormatter formatter;
 
-    public Review(@NotNull List<ReviewFile> files) {
+    public Review(@NotNull List<ReviewFile> files, Configuration configuration) {
```

Comment:
I'm with you on that, it feels pretty unnatural. Another possibility is pre-parsing the configuration and providing only the parameters it needs.

The "configurability" issue is something I'm dealing with in another project. I thought of enumerating all configuration options (much like the GeneralOption), annotating with an enum-set of the config args needed, and passing only those arguments in the constructor. But that gets messy rather quickly with reflection...

NB: a monkey with an IDE could have done all the same changes I did. I fixed the tests to use the new signatures, but I haven't "tested" it on any code reviews.